prov/tcp/src/tcpx_progress.c

```diff
@@ -204,30 +169,30 @@ static int process_srx_entry(struct tcpx_xfer_entry *rx_entry)
 		return ret;
 
 	if (ret) {
-		FI_WARN(&tcpx_prov, FI_LOG_DOMAIN,
+		FI_WARN(&tcpx_prov, FI_LOG_EP_DATA,
 			"msg recv Failed ret = %d\n", ret);
 
-		tcpx_ep_shutdown_report(rx_entry->ep,
-					&rx_entry->ep->util_ep.ep_fid.fid);
+		/* Is this check needed? srx did not have it */
+		if (ret == -FI_ENOTCONN) {
+			tcpx_ep_shutdown_report(rx_entry->ep,
+				&rx_entry->ep->util_ep.ep_fid.fid);
+		}
 	}
 
-	if ((ntohs(rx_entry->hdr.base_hdr.flags) &
-	     OFI_DELIVERY_COMPLETE) && !ret) {
+	/* srx included && !ret as part of check */
+	if (ntohs(rx_entry->hdr.base_hdr.flags) & OFI_DELIVERY_COMPLETE) {
 		if (tcpx_prepare_rx_entry_resp(rx_entry))
 			rx_entry->ep->cur_rx_proc_fn = tcpx_prepare_rx_entry_resp;
+	} else {
+		tcpx_cq_report_completion(rx_entry->ep->util_ep.rx_cq,
+					  rx_entry, -ret);
 
-		return FI_SUCCESS;
-	}
-
-	tcpx_cq_report_completion(rx_entry->ep->util_ep.rx_cq,
-				  rx_entry, -ret);
+		/* srx had this */
+		if (rx_entry->ep->cur_rx_entry == rx_entry)
+			rx_entry->ep->cur_rx_entry = NULL;
```

Comment:
for the rx path this happens in the _tcpx_xfer_entry_release_. So I guess it can be swept into _tcpx_srx_xfer_release_, if you want to.